src/metadata/EntityMetadata.ts

```diff
@@ -746,6 +746,14 @@ export class EntityMetadata {
         if (firstId === undefined || firstId === null || secondId === undefined || secondId === null)
             return false;
 
+        // Optimized version for the common case
+        if (
+          typeof firstId.id === "string" && Object.keys(firstId).length === 1 &&
```

Comment:
In the event `typeof secondId.id !== "string"` wouldn't it be faster to do the two `typeof` checks first, then `Object.keys`?

```js
if (
  typeof firstId.id === 'string' &&
  typeof secondId.id === 'string' &&
  Object.keys(firstId).length === 1 &&
  Object.keys(secondId).length === 1
) {
  return firstId.id === secondId.id;
}
```

Reply:
You mean because `typeof` should be faster than `Object.keys()` in general?

Reply:
Yes

Reply:
Agree, I've udpated the code, thx!